core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1569,7 +1569,7 @@ private[spark] object BlockManager {
 
     val blockManagers = new HashMap[BlockId, Seq[String]]
     for (i <- 0 until blockIds.length) {
-      blockManagers(blockIds(i)) = blockLocations(i).map(_.host)
+      blockManagers(blockIds(i)) = blockLocations(i).map(b => s"executor_${b.host}_${b.executorId}")
```

Comment:
The name of this method should be updated, `blockIdsToHosts` seems doesn't reflect your change.

Reply:
blockIdsToLocations ?

Reply:
Also you'd better using `ExecutorCacheTaskLocation#toString` here instead of manually writing the location hint, which will be more robust.